### ISSUE TITLE
Parsing of S3 credentials from VCAP_SERVICES fails for s3-cf-service-broker

### ISSUE THREAD

**Issue by mogul** [cloudfoundry-community/mattermost-cf-integrator#1]
TL;DR: When you push Mattermost with an S3 service binding that lacks a `uri` or `url` entry, it fails to start up.

The adapter [looks for an S3 service config key called `uri` or `url`](https://github.com/cloudfoundry-community/mattermost-cf-integrator/blob/46190bd4d815beae5c2277e05073138065796aac/mci/parser.go#L12). It then [parses out the various S3 parameters from the expected URI/URL](https://github.com/cloudfoundry-community/mattermost-cf-integrator/blob/master/mci/cloudidy_s3.go#L16-L34).

It's unclear what broker this was written to work with. However, it definitely doesn't work with the [`cloudfoundry-community/s3-cf-service-broker`](https://github.com/cloudfoundry-community/s3-cf-service-broker), which [explicitly provides the access values in `VCAP_SERVICES`](https://github.com/cloudfoundry-community/s3-cf-service-broker/blob/master/README.md#format-of-credentials).

The symptom of this problem that you see when you follow the directions is that your instance never starts.

**Comment by ArthurHlt** [comment]
@mogul you're absolutely right, I will make the changes and i will let you know when it will be ready.
Like I'm trying to follow versions from mattermost I will repush a 3.0.2 version.

I was in fact assuming that s3 service broker was following the same credentials as [p-riakcs](http://docs.pivotal.io/p-riakcs), I will probably make a pull request on [cloudfoundry-community/s3-cf-service-broker](https://github.com/cloudfoundry-community/s3-cf-service-broker) because uri and url when possible are the best way to automatically parse a service.

The [cloudfoundry-community/s3-cf-service-broker](https://github.com/cloudfoundry-community/s3-cf-service-broker) assume that's the host is aws amzon s3 which make applications  dependent to this broker and can't change to another s3 broker.

**Comment by ArthurHlt** [comment]
and it's done, you can download the release here: https://github.com/cloudfoundry-community/mattermost-cf-integrator/releases/download/v3.0.2/mattermost-cf.zip

**Comment by mogul** [comment]
Wow, lightspeed! Thanks a ton.

**Comment by mogul** [comment]
And: agreed about the URI being the most cross-broker-compatible way to expect the credentials... was just about to make an issue on the S3 broker myself!

**Comment by ArthurHlt** [comment]
and the PR was merged, thank you for your issue.
